docs/specifications/broker-authentication.md

```diff
@@ -0,0 +1,107 @@
+# Broker Authentication
+
+This document attempts to outline the high-level plan for adding authentication
+to the broker api routes.
+
+When talking to service brokers, the [Service Catalog uses basic auth](#service-catalog-basic-auth). The Ansible Service Broker will implement Basic Auth using a secret shared between the catalog and the broker. The authentication will be configurable allowing a user i.e. developer to disable it.
+
+The plan for this sprint is to do the following:
+
+* make auth routes configurable by enhancing the configuration
+```
+auth:
+   - type: basic | INSERT OTHER TYPE HERE
+   - enable: true | false
+```
+  * update the following files to reflect the new entry:
+      * [deploy-ansible-service-broker.template.yaml](https://github.com/openshift/ansible-service-broker/blob/master/templates/deploy-ansible-service-broker.template.yaml)
+      * [prep_local_devel_env.sh](https://github.com/openshift/ansible-service-broker/blob/master/scripts/prep_local_devel_env.sh) script
+      * [example-config.yaml](https://github.com/openshift/ansible-service-broker/blob/master/etc/example-config.yaml)
+      * [README.md](https://github.com/openshift/ansible-service-broker/blob/master/README.md)
+      * docs directory:
+          * [deployment.md](https://github.com/openshift/ansible-service-broker/blob/master/docs/deployment.md)
+          * [config.md](https://github.com/openshift/ansible-service-broker/blob/master/docs/config.md)
+          * [local_deployment.md](https://github.com/openshift/ansible-service-broker/blob/master/docs/local_deployment.md)l
+      * [`make deploy`](https://github.com/openshift/ansible-service-broker/blob/master/Makefile#L57-L58) does not need updating because it uses the above template.
+  * [catasb](https://github.com/fusor/catasb/tree/dev)
+      * pulls down the `deploy-ansible-service-broker.template.yaml`
+      * may have to update all.yml in case we need to override the auth setting
+
+* create a secret that both the broker and service catalog can use
+
+* secret rotation is desired but needs more research, see [#open-issues](#open-issues)
+
+* use [`http.BasicAuth`](https://golang.org/pkg/net/http/#Request.BasicAuth) to handle the basic auth case
```

Comment:
Yeah it's a way overloaded term. In the context of requests and auth, basically a hook in the incoming request stack that runs before the request hits the handler. Allows you to inspect the incoming req, authenticate (check auth params against known user, create login record, possibly enrich request with session data) and call some kind of next() that allows the request to pass through the rest of the stack, or reject() that will error out the request with a 401.